pkg/cosign/fetch.go

```diff
@@ -56,32 +60,45 @@ func FetchSignatures(ref name.Reference) ([]SignedPayload, *v1.Descriptor, error
 		return nil, nil, err
 	}
 
-	signatures := []SignedPayload{}
-	for _, desc := range m.Layers {
-		base64sig, ok := desc.Annotations[sigkey]
-		if !ok {
-			continue
-		}
-		l, err := sigImg.LayerByDigest(desc.Digest)
-		if err != nil {
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
Not a problem with this PR, but we should be plumbing a context through the entire library so that we can cancel stuff, e.g. https://golang.org/pkg/os/signal/#NotifyContext

Reply:
it was easy enough to plumb through so I just added it in